libvast/native-plugins/segment_store.cpp

```diff
@@ -147,24 +147,10 @@ caf::expected<uint64_t> handle_lookup(Actor& self, const vast::query& query,
       for (size_t i = 0; i < slices.size(); ++i) {
         const auto& slice = slices[i];
         const auto& checker = checkers[i];
-        if (extract.policy == query::extract::preserve_ids) {
-          for (auto& sub_slice : select(slice, ids)) {
-            if (query.expr == expression{}) {
-              self->send(extract.sink, sub_slice);
-              num_hits += sub_slice.rows();
-            } else {
-              auto hits = evaluate(checker, sub_slice);
-              num_hits += rank(hits);
-              for (auto& final_slice : select(sub_slice, hits))
```

Comment:
Is `select` now dead code?

Reply:
Almost, we still use it in `truncate` which is used in the sink. `truncate` could be changed easily though.

Reply:
truncate should just use `arrow::RecordBatch::Slice` instead.

Reply:
Same deal with `split`. So just `to_record_batch`, and then re-create a table slice from the same layout and the sliced record batch (we have a constructor for that already). Shouldn't be hard too write, so I think it'd fit in the scope of this PR.

Reply:
Ah damnit, so maybe an additional follow-up story for that then.